torch/_dynamo/variables/misc.py

```diff
@@ -767,16 +767,15 @@ def func(tx, item, acc):
             return variables.ListIteratorVariable(
                 items, mutable_local=MutableLocal(), **options
             )
-        elif (
-            self.value is functools.wraps
-            and not kwargs
-            and len(args) == 1
-            and args[0].source
-        ):
+        elif self.value is functools.wraps and not kwargs and len(args) == 1:
 
             def wraps(fn):
                 if isinstance(fn, variables.NestedUserFunctionVariable):
-                    return fn.clone(wraps_source=args[0].source)
+                    # `wraps_source` should implement `reconstruct`
+                    # which NestedUserFunctionVariable does.
+                    # We prefer `args[0].source` if that is available.
+                    wraps_source = args[0].source if args[0].source else fn
+                    return fn.clone(wraps_source=wraps_source)
```

Comment:
@voznesenskym could we get a sanity check here please?
- Previously, functools.wraps(f) support didn't work if f (a NestedUserFunctionVariable) had no source. (NestedUserFunctionVariable created in the current Dynamo context don't have a source.)
- As far as we can tell, f's source is used to reconstruct the `functools.wraps(f)` call during a graph break: the codegen uses the source of `f` to pull `f`, and then invokes functools.wraps on it.
- The new strategy is to reconstruct the `functools.wraps(f)` call by literally calling `f.reconstruct()` (since every NestedUserFunctionVariable has a reconstruct() method) and then calling functools.wraps on it.